Purchasing.Mvc/Controllers/WorkgroupController.cs

```diff
@@ -479,6 +480,125 @@ public async Task<ActionResult> AddAccount(int id, WorkgroupAccount workgroupAcc
             return View(viewModel);
         }
 
+        [HttpGet]
+        public ActionResult UpdateAccountsFromFinjector(int id)
+        {
+            var workgroup = _workgroupRepository.GetNullableById(id);
+            if (workgroup == null)
+            {
+                ErrorMessage = "Workgroup could not be found";
+                return this.RedirectToAction(nameof(WorkgroupController.Index), typeof(WorkgroupController).ControllerName(), new { showAll = false });
+            }
+
+            if (workgroup.Administrative)
+            {
+                ErrorMessage = "Accounts may not be added to an administrative workgroup.";
+                return this.RedirectToAction(nameof(Details), new { id = id });
+            }
+
+            return View(workgroup);
+        }
+
+        [HttpPost]
+        public ActionResult UpdateAccountsFromFinjector(int id, string chartsString, string onlystring, bool onlyAdd = false)
```

Comment:
Why not have .net do the serialization and take a param of the right type instead of the string which you manually parse later?

Reply:
If you mean like WorkgroupFinjectorChart[] chartsString I couldn't get the binding to work.